samples/mpsl/timeslot/src/main.c

```diff
@@ -119,30 +130,73 @@ static void mpsl_timeslot_demo(void)
 		return;
 	}
 
-	err = mpsl_timeslot_session_open(mpsl_timeslot_callback);
+	api_call = OPEN_SESSION;
+	err = k_msgq_put(&mpsl_api_msgq, &api_call, K_FOREVER);
 	if (err) {
 		error();
 	}
-	err = mpsl_timeslot_request(&timeslot_request_earliest);
+
+	api_call = MAKE_REQUEST;
+	err = k_msgq_put(&mpsl_api_msgq, &api_call, K_FOREVER);
 	if (err) {
 		error();
 	}
 
 	printk("Press any key to close the session.\n");
 	console_getchar();
 
-	err = mpsl_timeslot_session_close();
+	api_call = CLOSE_SESSION;
+	err = k_msgq_put(&mpsl_api_msgq, &api_call, K_FOREVER);
 	if (err) {
 		error();
 	}
 }
 
+/* MPSL API calls must be done from a non-preemptible thread.
```

Comment:
What about: "To ensure thread safe operation, call all MPSL APIs from a non-preemtible thread"

Reply:
Ok, I can replace the whole comment with this.